app/controllers/downgrade_controller.rb

```diff
@@ -0,0 +1,17 @@
+class DowngradeController < ApplicationController
+  def new
+  end
+
+  def create
+    current_user.update_attribute(:role, 0)
+    wikis = Wiki.all
```

Comment:
we should be more efficient here - if your app had thousands of users, this would be extremely costly to load all wikis and then update each one individually. Instead, we should just make public the wikis related to the specific user.

I envision something like:

```ruby
# DowngradeController
def create
  current_user.downgrade_user!
end
```

```
# User model
def downgrade_user!
  self.role = 0
  wikis.make_public! # this would be a method defined in your Wiki model
end
```